src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class SquareSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();

```

Comment:
It's better to inject dependencies through the constructor. This will make it easier to control dependencies and test the code.